.github/workflows/clang-compile-tests.yml

```diff
@@ -32,36 +32,41 @@ jobs:
   build-and-tests-on-ubuntu:
 
     runs-on: ubuntu-22.04
+    container:
+      image: ghcr.io/nextsimhub/nextsimdg-dev-env:latest
 
     steps:
     - uses: actions/checkout@v2
-    - name: installs
-      run: |
-        sudo apt-get update
-        sudo apt-get install netcdf-bin libnetcdf-c++4-dev libboost-all-dev libeigen3-dev cmake libnetcdff-dev libhdf5-dev
-        sudo apt-get install python3-netcdf4
-
-        svn checkout http://forge.ipsl.jussieu.fr/ioserver/svn/XIOS/trunk xios
-        cp .github/xios_arch/ubuntu.arch xios/arch/arch-GCC_LINUX.path
-        cp .github/xios_arch/ubuntu.env xios/arch/arch-GCC_LINUX.env
-        cp .github/xios_arch/ubuntu.fcm xios/arch/arch-GCC_LINUX.fcm
-        (cd xios && ./make_xios --arch GCC_LINUX --netcdf_lib netcdf4_seq)
 
-    - name: make
+    - name: build and compile
       run: |
-        mkdir -p build
-        cd build
-        cmake -DENABLE_MPI=ON -Dxios_DIR="$(pwd)/../xios" -DENABLE_XIOS=ON -DCMAKE_CXX_COMPILER="$(which mpic++)" ..
-        make
+        . /opt/spack-environment/activate.sh
+        mkdir -p build && cd build
+        cmake -DENABLE_MPI=ON -Dxios_DIR="/xios" -DENABLE_XIOS=ON -DCMAKE_CXX_COMPILER="$(which mpic++)" ..
+        make -j 4
+
     - name: run tests
       run: |
+        . /opt/spack-environment/activate.sh
+        apt install -y wget
+        cd build
+        (cd core/test && wget "ftp://ftp.nersc.no/nextsim/netCDF/partition_metadata_1.nc")
         for component in core physics
         do
             cd $component/test
             # run serial tests
-            for file in $(find test* -maxdepth 0 -type f | grep -vP "_MPI\d*$"); do echo $file; ./$file; done
-            # run MPI tests that require 2 ranks
-            for file in $(find test* -maxdepth 0 -type f | grep -P "_MPI2$"); do echo $file; mpirun -n 2 --oversubscribe ./$file; done
+            for file in $(find test* -maxdepth 0 -type f | grep -vP "_MPI\d*$")
+            do
+              echo $file
+              ./$file
+            done
+            # run MPI tests
```

Comment:
I chose not to split because we can reuse more of the code that way. I don't have a strong preference though. I thought this was just easier to maintain. As for the download it should take about 1s.